plugins/callback/mail.py

```diff
@@ -120,27 +120,34 @@ def mail(self, subject='Ansible error mail', body=None):
 
         smtp = smtplib.SMTP(self.smtphost, port=self.smtpport)
 
+        sender_address = email.utils.parseaddr(self.sender)
```

Comment:
You need to handle the case `self.sender is None`. It's probably best to just add `From: None\n` to `content` in that case.

Reply:
I’m not sure this is needed because this `mail` function is only called from `mail_result`.
https://github.com/ansible-collections/community.general/blob/77a0c139c9c929fb08d3b9c7eb3409b08bdb6857/plugins/callback/mail.py#L225

And `mail_result` is already defaulting `self.sender` when it’s `None`:
https://github.com/ansible-collections/community.general/blob/77a0c139c9c929fb08d3b9c7eb3409b08bdb6857/plugins/callback/mail.py#L163-L164

Reply:
Ah, thanks for pointing that out! Maybe it wasn't necessary to deprecate sender not being specified...